sql/core/src/main/scala/org/apache/spark/sql/execution/python/ArrowEvalPythonExec.scala

```diff
@@ -26,6 +26,28 @@ import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.execution.SparkPlan
 import org.apache.spark.sql.types.StructType
 
+private object BatchIterator {
+  class InnerIterator[T](iter: Iterator[T], batchSize: Int) extends Iterator[T] {
```

Comment:
could you just make `InnerIterator` a nested class in `BatchIterator` instead of here in the companion object?

Reply:
Done.